lib/livebook/intellisense/identifier_matcher.ex

```diff
@@ -207,6 +207,48 @@ defmodule Livebook.Intellisense.IdentifierMatcher do
     match_module_function(mod, hint, ctx) ++ match_module_type(mod, hint, ctx)
   end
 
+  defp match_struct_fields_or_local_or_var(hint, ctx) do
+    case expand_struct_fields(ctx) do
+      {:ok, struct, fields} ->
+        for field <- fields, do: {:map_field, field, nil}
```

Comment:
Is it okay to use `:map_field` here or should I add a new `:struct_field` type?

Reply:
I am not sure how it works for the Monaco editor, but IEx behaves like this:

Take a NaiveDateTime struct, it has both `month` and `minute`. If you type `m`, you want to show both `month` and `minute` as options. However, if you type `mo`, you want to type show only `month` as option and, if the user press tab, you want to autocomplete to `month: `. In order to encode the additional `: `, you need a different type than `map_field.

Reply:
Yeah, having a separate item here makes sense, especially that we need to add the default value. To avoid confusion with access syntax maybe something like `:in_struct_field` (alternatives welcome :see_no_evil:).

@mveytsman so you can add `{:in_struct_field, name(), default :: value()}` and then handle this new item in `Livebook.Intellisense`.

Reply:
My first pass was to rename `:map_field` to `:map_field_access` since it correctly handles both map and struct accesses, and call this `:struct_field`

Do you still prefer `:in_struct_field` @jonatanklosko ?

Reply:
Generally `:map_field_access` sounds good, but it doesn't fit to "identifier" that much. Similarly we have `:function` and not `:function_call`, so I'm leaning towards keeping `:map_field` as is :thinking:

Reply:
As an argument for it, `:map_field_access` is used in IEx, but I see that it doesn't fit identifier. I'll keep thinking on it as I refactor this to work right :)